### ISSUE TITLE
Additions to the CONTRIBUTING doc

### ISSUE THREAD

**Issue by carlisia** [vmware-tanzu/velero#1387]
1) Imports

For Kube/Ark imports, since many of the packages have names like `v1`, we often give them explicit names like `corev1`, `arkv1`, etc. to disambiguate. The problem is that we have no consistency in how we do this, so every file is different.

I'm proposing a standard way of naming these across Ark. The convention would be:

    <group><version><api | client | informer | ...>

So, for example:
```
    import (
        corev1api "k8s.io/api/core/v1"
    	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
    	corev1client "k8s.io/client-go/kubernetes/typed/core/v1"
    	corev1listers "k8s.io/client-go/listers/core/v1"
        
        velerov1api ""github.com/heptio/velero/pkg/apis/velero/v1"
        velerov1client "github.com/heptio/velero/pkg/generated/clientset/versioned/typed/velero/v1"
    )
```
Ref: https://github.com/heptio/velero/issues/494

2) Code

- Our standard is to capitalize log messages, but keep actual error messages lower-cased

~3) Protobuf~

~To generate protobuf files, must have version 1.0.0 of protoc-gen-go. Follow instructions in the link below and set git tag to: `GIT_TAG="v1.0.0"`
https://github.com/golang/protobuf#installation~

Protobuf generation was addressed here: https://github.com/heptio/velero/pull/1887

4) ...

**Comment by nrb** [comment]
For 1, we have https://github.com/heptio/velero/issues/494, which has kind of been an "as we go" thing. Maybe it's worth doing a pass just to clean these up?

**Comment by carlisia** [comment]
This issue is about documenting that this is even a thing. We know that but new contributors won't.

**Comment by skriss** [comment]
Let's definitely start to document this stuff. xref #867

**Comment by skriss** [comment]
xref #1104

**Comment by carlisia** [comment]
More things:

>  wrap/add a stack only to errors that are being directly returned from non-velero code (see: https://github.com/heptio/velero/pull/1616#discussion_r299526614)

> use the k8s.io/apimachinery/pkg/util/sets (see: https://github.com/heptio/velero/pull/1616#discussion_r299528781)

**Comment by carlisia** [comment]
More:

Generating mocks for our interfaces.

We use a package to do that. 

Example: if you want to change this mock: https://github.com/heptio/velero/blob/master/pkg/restic/mocks/backupper.go

Run:

```
go get github.com/vektra/mockery/.../
cd pkg/restic
mockery -name=Backupper
```

**Comment by skriss** [comment]
would be great to ship these docs updates as part of v1.1 if anyone has time -- though not a must-have.

**Comment by carlisia** [comment]
When running the websiet locally, this is the command that works for me:

`bundle exec jekyll serve --livereload --future`

Otherwise content that I'm adding is not included/displayed as expected.

**Comment by carlisia** [comment]
More instructions for the website.

When adding a blog post:

- The `author_name` and the `tags` fields must match. It should either be `Velero Team`, or the individual contributor's name (preferred). Ex: `author_name: Carlisia Campos`, `tags: ['Carlisia Campos'].

- If there is no image added to the header of the post, the default Velero logo will be used. This is fine, but not ideal. If there's an image that can be used as the blog post icon, the `image` field must be set to: `image: ../img/posts/<your_image_name.png>`. Important: this image file must be added to the `/site/img/posts` folder and also to the `/site/tags/img/posts` folder. If this last folder doesn't contain the image file, the image will be broken on the page for the author list of posts.

**Comment by carlisia** [comment]
Instructions to add: a changelog file for any significant change.

**Comment by carlisia** [comment]
Closed by: Add code and website guidelines #2032
